src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -285,12 +282,60 @@ await onMessage(messageContext)
             }
         }
 
+        async Task MovePoisonMessageToErrorQueue(Message message)
+        {
+            try
+            {
+                await sqsClient.SendMessageAsync(new SendMessageRequest
+                {
+                    QueueUrl = errorQueueUrl,
+                    MessageBody = message.Body
+                }).ConfigureAwait(false);
+                // The MessageAttributes on message are read-only attributes provided by SQS
+                // and can't be re-sent. Unfortunately all the SQS metadata 
+                // such as SentTimestamp is reset with this send.    
+            }
+            catch (Exception ex)
+            {
+                Logger.Error($"Error moving poison message to error queue at url {errorQueueUrl}. Moving back to input queue.", ex);
+                try
+                {
+                    await sqsClient.ChangeMessageVisibilityAsync(new ChangeMessageVisibilityRequest
+                    {
+                        QueueUrl = queueUrl,
+                        ReceiptHandle = message.ReceiptHandle,
+                        VisibilityTimeout = 0
+                    }).ConfigureAwait(false);
+                }
+                catch (Exception changeMessageVisibilityEx)
+                {
+                    Logger.Warn($"Error returning poison message back to input queue at url {queueUrl}. Poison message will become available at the input queue again after the visibility timeout expires.", changeMessageVisibilityEx);
+                }
+                return;
+            }
+
+            try
+            {
+                await sqsClient.DeleteMessageAsync(new DeleteMessageRequest
+                {
+                    QueueUrl = queueUrl,
+                    ReceiptHandle = message.ReceiptHandle
+                }).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Logger.Warn($"Error removing poison message from input queue {queueUrl}. This may cause duplicate poison messages in the error queue for this endpoint.", ex);
```

Comment:
To date the pump swallows all exceptions; there's no circuitbreaker in the transport (is there another circuitbreaker outside of the transport you're referring to?)